spec/features/user_show_spec.rb

```diff
@@ -0,0 +1,46 @@
+require 'rails_helper'
+
+RSpec.describe 'users/show.html.erb', type: :feature do
+  describe 'show page' do
+    let(:user) { User.create(name: 'Tom', photo: 'https://picsum.photos/id/64/200', bio: 'Teacher from Mexico', post_counter: 0) }
+    let!(:post1) { Post.create(user:, title: 'Post 1', text: 'This is my first post', author_id: 1) }
+    let!(:post2) { Post.create(user:, title: 'Post 2', text: 'This is my second post', author_id: 2) }
+    let!(:post3) { Post.create(user:, title: 'Post 3', text: 'This is my third post', author_id: 3) }
+
+    before do
+      visit user_path(1)
+    end
+
+    it "shows the user's profile picture" do
+      expect(page).to have_css("img[src*='#{user.photo}']")
+    end
+
+    it "shows the user's username" do
+      expect(page).to have_content(user.name)
+    end
+
+    it 'shows the number of posts the user has written' do
+      expect(page).to have_content(user.post_counter)
+    end
+
+    it "shows the user's bio" do
+      expect(page).to have_content('Bio')
+      expect(page).to have_content('Teacher from Mexico')
+    end
+
+    it "shows the user's first 3 posts" do
+      expect(page).to have_content('This is my first post')
+      expect(page).to have_content('First post from Lilly')
+      expect(page).to have_content('Third post')
+    end
```

Comment:
- [ ] Let's kindly make sure this tests display only the user's 3 most recent posts when the user create more than 4 posts. Only 3 most recent posts of the user should be expected to display :+1: